apps/astarte_housekeeping/lib/astarte_housekeeping/queries.ex

```diff
@@ -24,46 +24,92 @@ defmodule Astarte.Housekeeping.Queries do
 
   @default_replication_factor 1
 
+  def keyspace_from_realm(realm_name) do
+    customer_keyspace = Config.instance_id!()
+
+    case :persistent_term.get(customer_keyspace <> realm_name <> "encoded", nil) do
```

Comment:
But, but but! There is no need to store the `astarte_instance_id` as part  of the key in the persistent term storage, what you want is a map realm_name => keyspace_name.

So this function can become something like:
```elixir
def keyspace_from_name(realm_name) do
  case :persistent_term.get(realm_name, nil) do
    nil ->
      keyspace_name_base = Config.astarte_instance_id!() <> realm_name
      keyspace_name = some_clever_transformation_of(keyspace_name_base)
      :persistent_term.put(realm_name, keyspace_name)
    value -> value
  end
end
```
This will also apply to the `astarte` keyspace (even if it is a keyspace without realm)